src/ApimEventProcessor/HttpMessage.cs

```diff
@@ -29,44 +76,64 @@ public static HttpMessage Parse(Stream stream)
             }
         }
 
+        private static void TransformResponseHeaders(HttpResponseMessage response, string headerString) 
+        {
+            string[] splitHeaders = headerString.Split(new string[] { ";;" }, StringSplitOptions.None);
+
+            foreach (var h in splitHeaders)
+            {
+                var kv = h.Trim().Split(':');
+                if (kv[0].Trim() == "Content-Type") {
+                    response.Content.Headers.ContentType = new MediaTypeHeaderValue(kv[1].Trim());                   
```

Comment:
If `h` is blank/empty string, this will cause an unhandled exception?

Reply:
Added checks for empty string. MediaTypeHeaderValue accept string which will be set in Content-Tye header downstream. ResponseMessage doesn't allow to set Content-Type in the header, so we've to set it in Content Header. When preparing the EventResponseModel, we'd merge headers from both ResponseMessage and Content Headers, so Content-Type header will set and we'd able to parse data correctly.